homeassistant/components/abode/config_flow.py

```diff
@@ -0,0 +1,76 @@
+"""Config flow for the Abode Security System component."""
+import logging
+import voluptuous as vol
+from abodepy import Abode
+from abodepy.exceptions import AbodeAuthenticationException
+
+from homeassistant import config_entries
+from homeassistant.core import callback
+from homeassistant.const import CONF_PASSWORD, CONF_USERNAME
+
+from .const import DOMAIN  # pylint: disable=W0611
+
+CONF_POLLING = "polling"
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class AbodeFlowHandler(config_entries.ConfigFlow, domain=DOMAIN):
+    """Config flow for Abode."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_LOCAL_POLL
+
+    def __init__(self):
+        """Initialize."""
+        self.data_schema = {
+            vol.Required(CONF_USERNAME): str,
+            vol.Required(CONF_PASSWORD): str,
+        }
+
+    async def async_step_user(self, user_input=None):
+        """Handle a flow initialized by the user."""
+
+        if self._async_current_entries():
+            return self.async_abort(reason="single_instance_allowed")
+
+        if not user_input:
+            return self._show_form()
+
+        username = user_input[CONF_USERNAME]
+        password = user_input[CONF_PASSWORD]
+        polling = user_input[CONF_POLLING] if CONF_POLLING in user_input else False
+
+        try:
+            await self.hass.async_add_executor_job(Abode, username, password, True)
+
+        except AbodeAuthenticationException as error:
+            if error.errcode == 400:
+                return self._show_form({"base": "invalid_credentials"})
```

Comment:
When abode is configured using the configuration.yaml file, if the credentials are invalid, there is currently no way for the user to know. What shows up in the UI is abode is discovered and the user can click "Configure" to setup abode via the config flow.

I think adding some logging here that matches the config flow messages would be useful:

`_LOGGER.warning("Invalid credentials.")`

`_LOGGER.warning("Error occurred with Abode server.")`